Tasks/MavenV3/Tests/L0RestoreOriginalPomXml.ts

```diff
@@ -0,0 +1,148 @@
+import * as path from 'path';
+
+import { TaskMockRunner } from 'azure-pipelines-task-lib/mock-run';
+import { TaskLibAnswers } from 'azure-pipelines-task-lib/mock-answer';
+
+import { getTempDir, initializeTest, MavenTaskInputs, setInputs } from './TestUtils';
+
+const taskPath = path.join(__dirname, '..', 'maventask.js');
+
+const taskRunner = new TaskMockRunner(taskPath);
+
+// Common initial setup
+initializeTest(taskRunner);
+
+// Set Inputs
+const inputs: MavenTaskInputs = {
+    mavenVersionSelection: 'Default',
+    mavenPOMFile: 'pom.xml',
+    options: '',
+    goals: 'package',
+    javaHomeSelection: 'JDKVersion',
+    jdkVersion: 'default',
+    publishJUnitResults: true,
+    testResultsFiles: '**/TEST-*.xml',
+    mavenOpts: '-Xmx2048m',
+    checkstyleAnalysisEnabled: false,
+    pmdAnalysisEnabled: false,
+    findbugsAnalysisEnabled: false,
+    mavenFeedAuthenticate: true,
+    codeCoverageTool: 'JaCoCo',
+    restoreOriginalPomXml: true
+};
+setInputs(taskRunner, inputs);
+
+// Set up environment variables (task-lib does not support mocking getVariable)
+// Env vars in the mock framework must replace '.' with '_'
+delete process.env.M2_HOME; // Remove in case process running this test has it already set
```

Comment:
Do we really need it?

Reply:
I think so. If we don't have this line and the user has M2_HOME env variable set on his machine then the test might fail since the task might call maven from unexpected path

Reply:
Is it mean that this could potentially break my local setup for maven? May be we could mock this variable in any way?

Reply:
No, this will not break your local setup since deleting properties from process.env will not change your system's ENV variables